lfshttp/standalone/standalone.go

```diff
@@ -244,6 +245,15 @@ func (h *fileHandler) download(oid string, size int64) (string, string, error) {
 		return oid, "", err
 	}
 
+	// Set the timestamp so if we hard link this file into a temporary
+	// location, our temporary files don't get deleted because they're too
+	// old.
+	now := time.Now()
+	err = os.Chtimes(src, now, now)
```

Comment:
I'm a bit worried by performance effect of such change. Unfortunately, if `lfs.LinkOrCopy` ends up doing `os.Link`, we don't get a change to change mtime.

On the other side, if we're able to do `os.Link(src, dst)` we don't need temp file at all!

So, I believe a better fix is:

1. Try to do `os.Link(src, dst)`. If it succeeded (either directly or via a race condition where a parallel process managed to create `dst` earlier than us), end.
2. Create temp, copy src -> temp
3. `os.Rename(temp, dst)`.

But it looks like current level of abstraction doesn't provide us enough context in order to perform `os.Link(src, dst)`?

Reply:
We do need the temporary file regardless, because the standalone transfer protocol doesn't let us write directly into the LFS storage.  We have to provide a download location for the file so that the main process can copy or link it into the repository.

Reply:
That's unfortunate. Direct `os.Link` from src to LFS storage would kill two birds with one stone - no need to write to src and no issues with temp file mtime.

Reply:
I just realized this won't work if the remote repository is read only or otherwise doesn't have permissions.  I'll need to revise this.

Reply:
Fixed.